### PR TITLE
Fix `cupyx.special.gammainc`, `lpmv` and `sph_harm` for hip

### PR DESCRIPTION
**Pull request by emcastillo**
Fixing more hip stuff

### CODE REVIEW

File: cupyx/scipy/special/_gammainc.py [hunk 1]
```diff
@@ -397,7 +397,7 @@
 #define d_K 25
 #define d_N 25
 
-static const double d[d_K][d_N] =
+__device__ static const double d[d_K][d_N] =
```
Comment (grlee77) [inline cupyx/scipy/special/_gammainc.py:400]: oops, I had done this elsewhere in other PRs, but missed this one. thanks!

**Review by grlee77 — APPROVED** [review]
Thanks @emcastillo, LGTM

**Review by kmaehashi — APPROVED** [review]

**Comment by kmaehashi** [comment]
Backport not needed as this is v11 only (#5687).

**Pull request merged**